utils/url_validation.ts

```diff
@@ -0,0 +1,32 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+export function isValidUrl(string: string): boolean {
+  try {
+    const { protocol } = new URL(string);
+    const pattern = /^(https?):/
+
+    return pattern.test(protocol);
+  } catch (_) {
+    return false;
+  }
```

Comment:
```suggestion
  return URL.canParse(string) && string.startsWith("http");
```
I prefer to avoid regex when possible.

Reply:
nice, learned something new!

Reply:
I avoided using `URL.canParse()` because types are missing in the runtime. Fix incoming: https://github.com/denoland/deno/pull/20244